### PR TITLE
Compare labels

### PR DESCRIPTION
**Pull request by jo-mueller**
# Description

I added a method to compare labels between two label images label_image_x and label_image_y. The difference between the images is quantified in terms of Jaccard score and Dice score. The jaccard_score uses the function implemented in scikit-learn, the dice score is derived from the Jaccard-score as described in [Wikipedia](https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient#Difference_from_Jaccard). 

I added a notebook that visualizes the label-wise differences based on example data that is already shipped with biapol_utilities. Tests are included.

## Type of change

* [ ] Bug-fix
* [x] New feature
* [ ] Breaking change
* [ ] Documentation update

## References 

Related to #17 

## Tests

* [ ] I adapted existing tests, because
* [x] I added new tests to cover the code change
* [x] All tests pass with my change

## Final checks

* [x] My change is the minimal possible work for the desired feature/fix
* [x] I updated the documentation where necessary to cover the change

### CODE REVIEW

File: biapol_utilities/label/_compare_labels.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    label_image_x = label_image_x.flatten()
+    label_image_y = label_image_y.flatten()
+
+    if not label_image_x.shape == label_image_y.shape:
+        raise Exception('Input images must have same dimension but have '
+                        f'{label_image_x.shape} and {label_image_y.shape}')
+
+    # get list of present labels in bothn images and allocate results
+    labels = np.unique(np.hstack([label_image_x, label_image_y]))
+    df = pd.DataFrame(columns=['label', 'jaccard_score'])
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:42]: Is there `dice_score` missing here?

File: biapol_utilities/label/_compare_labels.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    Parameters
+    ----------
+    label_image_x : ND-array, int
+        Ground truth label image of arbitrary dimensions.
+    label_image_y : ND-array, int
+        Predicted label image which will be compared to label_image_x.
+
+    Returns
+    -------
+    pandas dataframe with columns `[label, jaccard_score, dice_score]. Each row
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:27]:
```suggestion
    pandas DataFrame with columns `['label', 'jaccard_score', 'dice_score']`. Each row
```

File: biapol_utilities/label/_compare_labels.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    The Jaccard-score is defined as the intersection over union of two labelled
+    images.
+    The Dice score S can be derived from the Jaccard-score J through the
+    following relation:
+    .. math:: S = \frac{2J}{1+J}.
+
+    Parameters
+    ----------
+    label_image_x : ND-array, int
+        Ground truth label image of arbitrary dimensions.
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:21]:
```suggestion
        label image of arbitrary dimensions.
```

File: biapol_utilities/label/_compare_labels.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    label_image_y : ND-array, int
+        Predicted label image which will be compared to label_image_x.
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:23]:
```suggestion
        label image which will be compared to `label_image_x`. It should have the same dimensions.
```

File: biapol_utilities/label/_compare_labels.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from sklearn.metrics import jaccard_score
+import pandas as pd
+
+
+def compare_labels(label_image_x, label_image_y):
+    """
+    Compares two label images label_image_x and label_image_y regarding the
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:10]:
```suggestion
    Compares two label images to determine the
```

File: biapol_utilities/label/_compare_labels.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    label-wise Jaccard- and Dice scores.
+    The Jaccard-score is defined as the intersection over union of two labelled
+    images.
+    The Dice score S can be derived from the Jaccard-score J through the
+    following relation:
+    .. math:: S = \frac{2J}{1+J}.
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:16]:
```suggestion
    .. math:: S = \frac{2J}{1+J}.
    
    See also
    --------
    .. [1] https://en.wikipedia.org/wiki/Jaccard_index
```

File: biapol_utilities/label/_compare_labels.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        Predicted label image which will be compared to label_image_x.
+
+    Returns
+    -------
+    pandas dataframe with columns `[label, jaccard_score, dice_score]. Each row
+    corresponds to the measured quantitiy (Jaccard- or Dice score) of the
+    respective label.
+    """
+
+    # Flatten input
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:32]: Can you please write WHY what is done? "Flatten" is a term that means something different in ImageJ

File: biapol_utilities/label/_compare_labels.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+from sklearn.metrics import jaccard_score
+import pandas as pd
+
+
+def compare_labels(label_image_x, label_image_y):
+    """
+    Compares two label images label_image_x and label_image_y regarding the
+    label-wise Jaccard- and Dice scores.
+    The Jaccard-score is defined as the intersection over union of two labelled
+    images.
```
Comment (haesleinhuepf) [inline biapol_utilities/label/_compare_labels.py:13]:
```suggestion
    images [1].
```

**Review by haesleinhuepf — APPROVED** [review]
Hi Johannes @jo-mueller ,

just some minor comments.

Otherwise it looks great and you're welcome to go ahead and merge.

File: biapol_utilities/label/_compare_labels.py (hunk 1 above)
Comment (jo-mueller) [reply biapol_utilities/label/_compare_labels.py:42]: Indeed - changed in 51c42c48694db89a6f0e1bcd5a058597a28a3aa7

Comment (jo-mueller) [reply biapol_utilities/label/_compare_labels.py]: Indeed! Changed in 51c42c48694db89a6f0e1bcd5a058597a28a3aa7

File: biapol_utilities/label/_compare_labels.py (hunk 7 above)
Comment (jo-mueller) [reply biapol_utilities/label/_compare_labels.py]: Added this to the comment and replaced `flatten()` with `ravel()`:

```python
  # Convert image to 1D array. Makes elemt-wise comparison of arrays simpler.
  label_image_x = label_image_x.ravel()
  label_image_y = label_image_y.ravel()
```
